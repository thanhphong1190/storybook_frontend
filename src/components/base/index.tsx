export { Button, ButtonLink } from './Button'
export { SVGIcon, IconEnums } from './icon/svgIcon'
export { Icon, Icons } from './Icon'
