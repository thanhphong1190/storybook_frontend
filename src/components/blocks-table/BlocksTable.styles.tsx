import styled from 'styled-components'
import tw from 'twin.macro'

export const Section = styled.section`
  ${tw`w-full`}
`

export const Container = styled.div`
  ${tw`container mx-auto`}
`

export const BlocksWrapper = styled.div`
  ${tw`w-full flex flex-wrap items-stretch justify-center`}
`
export const Block = styled.div<{
  color?: 'blue' | 'white' | 'orange'
  isFull?: boolean
}>`
  ${tw`w-full flex flex-row items-stretch justify-between p-5`}
  ${({ isFull }) => (isFull ? tw`uppercase` : tw`md:w-1/3`)}

  ${({ color }) => {
    switch (color) {
      case 'blue':
        return `&{
                background-color: #01478c;
                background: -webkit-gradient(linear,left top,right top,from(#164cab),to(#0c1e54));
                background: linear-gradient(to right,#164cab,#0c1e54);
                color:#fff;
            }`

      case 'white':
        return tw`bg-white text-blue`
      case 'orange':
        return tw`bg-orange text-white`
      default:
        return tw`bg-white text-blue`
    }
  }}
`
export const BlockNumber = styled.div`
  ${tw`flex-initial flex items-center`}
`
export const TextWrapper = styled.div`
  ${tw`flex-1 flex items-center pl-4`}
`

export const Text = styled.div`
  ${tw`uppercase text-xl md:text-4xl`}
`
