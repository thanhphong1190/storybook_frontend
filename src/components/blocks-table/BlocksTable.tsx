import * as S from './BlocksTable.styles'
import * as G from '../../styles/global.styles'

interface Block {
  text: string
  isFull?: boolean
}
export interface BlocksTableProps {
  blocks: Block[]
}

export const BlocksTable = (props: BlocksTableProps) => {
  const { blocks } = props
  return (
    <S.Section>
      <S.Container>
        <S.BlocksWrapper>
          {blocks &&
            blocks.map((block, i) => (
              <S.Block
                key={i}
                color={block.isFull ? 'orange' : i % 2 == 1 ? 'blue' : 'white'}
                isFull={block.isFull}
              >
                <S.BlockNumber>
                  <G.Text
                    color={block.isFull ? 'white' : 'orange'}
                    size='6xl'
                    fontWeight='bold'
                  >
                    {i + 1}
                  </G.Text>
                </S.BlockNumber>
                <S.TextWrapper>
                  {block.isFull ? (
                    <S.Text>{block.text}</S.Text>
                  ) : (
                    <G.Text size={block.isFull ? '4xl' : 'md'}>
                      {block.text}
                    </G.Text>
                  )}
                </S.TextWrapper>
              </S.Block>
            ))}
        </S.BlocksWrapper>
      </S.Container>
    </S.Section>
  )
}
