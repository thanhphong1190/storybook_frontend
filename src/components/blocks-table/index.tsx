export { BlocksTable, BlocksTableProps } from './BlocksTable'
