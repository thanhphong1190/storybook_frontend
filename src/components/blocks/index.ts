export { ContentBlock, ContentBlocksProps } from './ContentBlock'
export { AuthorBlock, AuthorBlockProps } from './AuthorBlock'
export { VideoBlock, PlayButton, VideoBlocksProps, VideoPlayer, VideoPlayerProps } from './VideoBlock'