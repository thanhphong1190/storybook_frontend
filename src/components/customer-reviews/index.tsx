export { default as ReviewCard } from './ReviewCard'
export { default as CustomerReviews } from './CustomerReviews'
