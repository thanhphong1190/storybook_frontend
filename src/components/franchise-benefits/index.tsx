export { FranchiseBenefits, FranchiseBenefitsProps } from './FranchiseBenefits'
export {
  FranchiseBenefitCard,
  FranchiseBenefitCardProps
} from './FranchiseBenefitCard'
