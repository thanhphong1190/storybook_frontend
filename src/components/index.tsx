export { ContentBlock } from './blocks/ContentBlock'
export { Header } from './header/Header'
export { Navigation } from './navigation/Navigation'
export { HeroImage } from './hero/HeroImage'
export { KeyMessage } from './key-message/KeyMessage'
export { Footer } from './footer/Footer'
export { default as Banner } from './banner/Banner'
export { default as OrangeBanner } from './orange-banner/OrangeBanner'
export { ReviewCard, CustomerReviews } from './customer-reviews'
export { StandardContent } from './standard-content'
export { FranchiseBenefitCard, FranchiseBenefits } from './franchise-benefits'
export { Hero } from './hero/Hero'
export { ReadyStartedBlocks } from './ready-started-blocks'
export { BlocksTable } from './blocks-table'
