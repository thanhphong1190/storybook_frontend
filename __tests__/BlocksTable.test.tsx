import { BlocksTable, BlocksTableProps } from '../src/components/blocks-table'
import renderer from 'react-test-renderer'

const props: BlocksTableProps = {
  blocks: [
    {
      text: 'You enquire about a Plus Fitness 24/7 Franchise'
    },
    {
      text: 'We call you to discuss your completed form'
    },
    {
      text:
        'We provide you with our Budgeting and Projections Tool and full disclosure document'
    },
    {
      text:
        'After your Q&A Session we confirm your chosen territory is both suitable and available'
    },
    {
      text: 'Support in selecting and negotiating on a suitable premises'
    },
    {
      text: 'DA prepared and then submitted to Council'
    },
    {
      text: 'Franchisee Induction Training program'
    },
    {
      text:
        'Building works. At the same time you will also start your Full Presale of Memberships.'
    },
    {
      text:
        'Our team will complete your aesthetic fit out and install your gym equipment'
    },
    {
      text: 'YOUR PLUS FITNESS 24/7 GRAND OPENING!',
      isFull: true
    }
  ]
}

it('BlocksTable renders correctly', () => {
  const tree = renderer.create(<BlocksTable {...props} />).toJSON()
  expect(tree).toMatchSnapshot()
})
