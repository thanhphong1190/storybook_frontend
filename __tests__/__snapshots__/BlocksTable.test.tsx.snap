// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlocksTable renders correctly 1`] = `
<section
  className="BlocksTablestyles__Section-sc-1j09n0t-0 cqBHAs"
>
  <div
    className="BlocksTablestyles__Container-sc-1j09n0t-1 gCrCNi"
  >
    <div
      className="BlocksTablestyles__BlocksWrapper-sc-1j09n0t-2 jHKEPm"
    >
      <div
        className="BlocksTablestyles__Block-sc-1j09n0t-3 ZCdEx"
        color="white"
      >
        <div
          className="BlocksTablestyles__BlockNumber-sc-1j09n0t-4 ksADlf"
        >
          <div
            className="globalstyles__Text-sc-1nc69wi-9 hJbKKz"
            color="orange"
            fontWeight="bold"
            size="6xl"
          >
            1
          </div>
        </div>
        <div
          className="BlocksTablestyles__TextWrapper-sc-1j09n0t-5 cHJOLd"
        >
          <div
            className="globalstyles__Text-sc-1nc69wi-9 hVBppN"
            size="md"
          >
            You enquire about a Plus Fitness 24/7 Franchise
          </div>
        </div>
      </div>
      <div
        className="BlocksTablestyles__Block-sc-1j09n0t-3 hoDlsa"
        color="blue"
      >
        <div
          className="BlocksTablestyles__BlockNumber-sc-1j09n0t-4 ksADlf"
        >
          <div
            className="globalstyles__Text-sc-1nc69wi-9 hJbKKz"
            color="orange"
            fontWeight="bold"
            size="6xl"
          >
            2
          </div>
        </div>
        <div
          className="BlocksTablestyles__TextWrapper-sc-1j09n0t-5 cHJOLd"
        >
          <div
            className="globalstyles__Text-sc-1nc69wi-9 hVBppN"
            size="md"
          >
            We call you to discuss your completed form
          </div>
        </div>
      </div>
      <div
        className="BlocksTablestyles__Block-sc-1j09n0t-3 ZCdEx"
        color="white"
      >
        <div
          className="BlocksTablestyles__BlockNumber-sc-1j09n0t-4 ksADlf"
        >
          <div
            className="globalstyles__Text-sc-1nc69wi-9 hJbKKz"
            color="orange"
            fontWeight="bold"
            size="6xl"
          >
            3
          </div>
        </div>
        <div
          className="BlocksTablestyles__TextWrapper-sc-1j09n0t-5 cHJOLd"
        >
          <div
            className="globalstyles__Text-sc-1nc69wi-9 hVBppN"
            size="md"
          >
            We provide you with our Budgeting and Projections Tool and full disclosure document
          </div>
        </div>
      </div>
      <div
        className="BlocksTablestyles__Block-sc-1j09n0t-3 hoDlsa"
        color="blue"
      >
        <div
          className="BlocksTablestyles__BlockNumber-sc-1j09n0t-4 ksADlf"
        >
          <div
            className="globalstyles__Text-sc-1nc69wi-9 hJbKKz"
            color="orange"
            fontWeight="bold"
            size="6xl"
          >
            4
          </div>
        </div>
        <div
          className="BlocksTablestyles__TextWrapper-sc-1j09n0t-5 cHJOLd"
        >
          <div
            className="globalstyles__Text-sc-1nc69wi-9 hVBppN"
            size="md"
          >
            After your Q&A Session we confirm your chosen territory is both suitable and available
          </div>
        </div>
      </div>
      <div
        className="BlocksTablestyles__Block-sc-1j09n0t-3 ZCdEx"
        color="white"
      >
        <div
          className="BlocksTablestyles__BlockNumber-sc-1j09n0t-4 ksADlf"
        >
          <div
            className="globalstyles__Text-sc-1nc69wi-9 hJbKKz"
            color="orange"
            fontWeight="bold"
            size="6xl"
          >
            5
          </div>
        </div>
        <div
          className="BlocksTablestyles__TextWrapper-sc-1j09n0t-5 cHJOLd"
        >
          <div
            className="globalstyles__Text-sc-1nc69wi-9 hVBppN"
            size="md"
          >
            Support in selecting and negotiating on a suitable premises
          </div>
        </div>
      </div>
      <div
        className="BlocksTablestyles__Block-sc-1j09n0t-3 hoDlsa"
        color="blue"
      >
        <div
          className="BlocksTablestyles__BlockNumber-sc-1j09n0t-4 ksADlf"
        >
          <div
            className="globalstyles__Text-sc-1nc69wi-9 hJbKKz"
            color="orange"
            fontWeight="bold"
            size="6xl"
          >
            6
          </div>
        </div>
        <div
          className="BlocksTablestyles__TextWrapper-sc-1j09n0t-5 cHJOLd"
        >
          <div
            className="globalstyles__Text-sc-1nc69wi-9 hVBppN"
            size="md"
          >
            DA prepared and then submitted to Council
          </div>
        </div>
      </div>
      <div
        className="BlocksTablestyles__Block-sc-1j09n0t-3 ZCdEx"
        color="white"
      >
        <div
          className="BlocksTablestyles__BlockNumber-sc-1j09n0t-4 ksADlf"
        >
          <div
            className="globalstyles__Text-sc-1nc69wi-9 hJbKKz"
            color="orange"
            fontWeight="bold"
            size="6xl"
          >
            7
          </div>
        </div>
        <div
          className="BlocksTablestyles__TextWrapper-sc-1j09n0t-5 cHJOLd"
        >
          <div
            className="globalstyles__Text-sc-1nc69wi-9 hVBppN"
            size="md"
          >
            Franchisee Induction Training program
          </div>
        </div>
      </div>
      <div
        className="BlocksTablestyles__Block-sc-1j09n0t-3 hoDlsa"
        color="blue"
      >
        <div
          className="BlocksTablestyles__BlockNumber-sc-1j09n0t-4 ksADlf"
        >
          <div
            className="globalstyles__Text-sc-1nc69wi-9 hJbKKz"
            color="orange"
            fontWeight="bold"
            size="6xl"
          >
            8
          </div>
        </div>
        <div
          className="BlocksTablestyles__TextWrapper-sc-1j09n0t-5 cHJOLd"
        >
          <div
            className="globalstyles__Text-sc-1nc69wi-9 hVBppN"
            size="md"
          >
            Building works. At the same time you will also start your Full Presale of Memberships.
          </div>
        </div>
      </div>
      <div
        className="BlocksTablestyles__Block-sc-1j09n0t-3 ZCdEx"
        color="white"
      >
        <div
          className="BlocksTablestyles__BlockNumber-sc-1j09n0t-4 ksADlf"
        >
          <div
            className="globalstyles__Text-sc-1nc69wi-9 hJbKKz"
            color="orange"
            fontWeight="bold"
            size="6xl"
          >
            9
          </div>
        </div>
        <div
          className="BlocksTablestyles__TextWrapper-sc-1j09n0t-5 cHJOLd"
        >
          <div
            className="globalstyles__Text-sc-1nc69wi-9 hVBppN"
            size="md"
          >
            Our team will complete your aesthetic fit out and install your gym equipment
          </div>
        </div>
      </div>
      <div
        className="BlocksTablestyles__Block-sc-1j09n0t-3 jBynHr"
        color="orange"
      >
        <div
          className="BlocksTablestyles__BlockNumber-sc-1j09n0t-4 ksADlf"
        >
          <div
            className="globalstyles__Text-sc-1nc69wi-9 jagBpF"
            color="white"
            fontWeight="bold"
            size="6xl"
          >
            10
          </div>
        </div>
        <div
          className="BlocksTablestyles__TextWrapper-sc-1j09n0t-5 cHJOLd"
        >
          <div
            className="BlocksTablestyles__Text-sc-1j09n0t-6 bNtgqp"
          >
            YOUR PLUS FITNESS 24/7 GRAND OPENING!
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;
